pytensor/tensor/interpolate.py

```diff
@@ -0,0 +1,197 @@
+        - 'linear': Linear interpolation
+        - 'nearest': Nearest neighbor interpolation
+        - 'first': Stepwise interpolation using the closest value to the left of the query point
+        - 'last': Stepwise interpolation using the closest value to the right of the query point
+        - 'mean': Stepwise interpolation using the mean of the two closest values to the query point
+    left_pad: TensorLike, optional
+        Value to return inputs `x_hat < x[0]`. Default is `y[0]`. Ignored if ``extrapolate == True``; in this
+        case, values `x_hat < x[0]` will be extrapolated from the endpoints of `x` and `y`.
+    right_pad: TensorLike, optional
+        Value to return for inputs `x_hat > x[-1]`. Default is `y[-1]`. Ignored if ``extrapolate == True``; in this
+        case, values `x_hat > x[-1]` will be extrapolated from the endpoints of `x` and `y`.
+    extrapolate: bool
+        Whether to extend the request interpolation function beyond the range of the input-output pairs specified in
+        `x` and `y.` If False, constant values will be returned for such inputs.
+
+    Returns
+    -------
+    interpolation_func: OpFromGraph
+        A function that can be used to interpolate new data. The function takes a single input `x_hat` and returns
+        the interpolated value `y_hat`. The input `x_hat` must be a 1d array.
+
+    """
+    x = as_tensor_variable(x)
+    y = as_tensor_variable(y)
+
+    sort_idx = argsort(x)
+    x = x[sort_idx]
+    y = y[sort_idx]
+
+    if left_pad is None:
+        left_pad = y[0]
+    else:
+        left_pad = as_tensor_variable(left_pad)
+    if right_pad is None:
+        right_pad = y[-1]
+    else:
+        right_pad = as_tensor_variable(right_pad)
+
+    x_hat = scalar("x_hat", dtype=x.dtype)
+    idx = searchsorted(x, x_hat)
+
+    if x.ndim != 1 or y.ndim != 1:
+        raise ValueError("Inputs must be 1d")
+
+    if method == "linear":
+        y_hat = _linear_interp1d(
+            x, y, x_hat, idx, left_pad, right_pad, extrapolate=extrapolate
+        )
+    elif method == "nearest":
+        y_hat = _nearest_neighbor_interp1d(
+            x, y, x_hat, idx, left_pad, right_pad, extrapolate=extrapolate
+        )
+    elif method == "first":
+        y_hat = _stepwise_first_interp1d(
+            x, y, x_hat, idx, left_pad, right_pad, extrapolate=extrapolate
+        )
+    elif method == "mean":
+        y_hat = _stepwise_mean_interp1d(
+            x, y, x_hat, idx, left_pad, right_pad, extrapolate=extrapolate
+        )
+    elif method == "last":
+        y_hat = _stepwise_last_interp1d(
+            x, y, x_hat, idx, left_pad, right_pad, extrapolate=extrapolate
+        )
+    else:
+        raise NotImplementedError(
+            f"Unknown interpolation method: {method}. "
+            f"Did you mean {get_close_matches(method, valid_methods)}?"
+        )
+
+    return Blockwise(
+        OpFromGraph(inputs=[x_hat], outputs=[y_hat], inline=False), signature="()->()"
```

Comment:
Any reason to jump directly to OpFromGraph and without inline? The pure PyTensor stuff would be core graph -> vectorize

When you OFG you forbid rewrites at the boundary of outside and inside OFG (as well merging of computations that are identical across different OFG).

For pad you were dispatching directly for JAX but here is not the case?

Reply:
Yeah I think it makes more sense to drop the OFG and return wrapper functions. One moment.